libutils/hash.c

```diff
@@ -257,15 +257,15 @@ Hash *HashNewFromKey(const RSA *rsa, HashMethod method)
     }
 
     unsigned char buffer[buf_len];
-    int actlen;
+    size_t actlen;
 
     actlen = BN_bn2bin(n, buffer);
-    CF_ASSERT(actlen <= buf_len, "Buffer overflow n, %d > %zu!",
+    CF_ASSERT(actlen <= buf_len, "Buffer overflow n, %ld > %zu!",
```

Comment:
The format for `size_t` is `%zd` IIRC.

Reply:
`size_t` is unsigned, so should be `%zu`;

https://github.com/cfengine/core/blob/master/CONTRIBUTING.md#string-formatting

Reply:
Correct, good catch. :+1: